packages/ButtonGroup/ButtonGroup.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import Box from '@tds/core-box'
+import Text from '@tds/core-text'
+import { componentWithName } from '@tds/util-prop-types'
+
+import ButtonGroupItem from './ButtonGroupItem/ButtonGroupItem'
+
+import safeRest from '../../shared/utils/safeRest'
+import { warn } from '../../shared/utils/warn'
+
+import styles from './ButtonGroup.modules.scss'
+
+/**
+ * An input component utilizing buttons that act as radios.
+ * @version ./package.json
+ */
+const ButtonGroup = ({ name, onChange, onFocus, onBlur, value, label, children, ...rest }) => {
+  const passedButtons = React.Children.map(children, child =>
+    React.cloneElement(child, {
+      name,
+      onChange,
+      onFocus,
+      onBlur,
+      checked: value === child.props.value,
+    })
+  )
+
+  const buttonValues = []
+  Object.keys(passedButtons).forEach(key => {
+    buttonValues.push(passedButtons[key].props.value)
+  })
+
+  if (buttonValues.indexOf(value) === -1) {
+    warn(
+      'ButtonGroup',
+      `Selected value "${value}" of ButtonGroup named "${name}" does not match the value of any button in the group. A button must be selected by default. Available button values are: ${buttonValues}`
+    )
+  }
+  return (
+    <fieldset name={name} {...safeRest(rest)}>
```

Comment:
Though this is inconsequential, we should put `safeRest` before other props for consistency among other components. In the case here, a user cannot override the `name` prop as used here since it matches the `name` html attribute, so `rest` props wouldn't be able to override it.